src/routes/authorize.rs

```diff
@@ -31,29 +35,45 @@ async fn authorize_handler(
         .ok_or(Error::OauthInvalidClientId)?
         .redirect_urls;
 
-    let are_requirements_empty = &app_state
+    let requirements_empty = &app_state
         .client_configs
         .get(&query.client_id)
         .ok_or(Error::OauthInvalidClientId)?
         .requirements
         .is_empty();
 
-    let is_redirect_uri_in_query = redirect_urls.contains(&query.redirect_uri);
+    // Support Authorization Code Flow and Implicit Flow.
+    let supported_response_types = ["id_token", "id_token token", "code"];
```

Comment:
These can and should also be constants, as they pop up in different places across the codebase.

Reply:
It could actually even be an enum, with functions to verify if a string has a valid value.